code/API_definitions/qod-provision.yaml

```diff
@@ -0,0 +1,876 @@
+        - `status` as `UNAVAILABLE` and
+        - `statusInfo` as `DELETE_REQUESTED`
+        There will be no notification event if the `status` was already `UNAVAILABLE`.
+      operationId: deleteProvision
+      parameters:
+        - $ref: "#/components/parameters/provisionId"
+        - $ref: "#/components/parameters/x-correlator"
+      responses:
+        "204":
+          description: Provision deleted
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+        "202":
+          description: Deletion request accepted to be processed. It applies for an async deletion process. `status` in the response will be `AVAILABLE` with `statusInfo` set to `DELETE_REQUESTED`.
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProvisionInfo"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+      security:
+        - openId:
+            - "qod-provision:provisions:delete"
+
+  /provisions/retrieve:
+    post:
+      tags:
+        - QoD Provision
+      summary: Gets the QoD provision for a given device
+      description: |
+        Retrieves the QoD provision for a given device.
+      operationId: retrieveProvision
+      parameters:
+        - $ref: "#/components/parameters/x-correlator"
+      requestBody:
+        description: Parameters to retrieve a provision by device
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RetrieveProvision"
+        required: true
+      responses:
+        "200":
+          description: Returns information about an active provision for the device. If no provision is found for the device, an error response 404 is returned with code "PROVISION_NOT_FOUND".
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProvisionInfo"
+              examples:
+                PROVISION_AVAILABLE:
+                  $ref: "#/components/examples/PROVISION_AVAILABLE"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/RetrieveProvision404"

```

Comment:
What about a second response to 404 for device not found?  Since this code implies that there isn't an active provision, which is technically true.  There are no active provisions on a none existent device, but the implications are different.

Reply:
The response (schema in components) already considers 2 codes: PROVISION_NOT_FOUND and DEVICE_NOT_FOUND. As I said above there is an error in the description and provision could be in any status.